### PR TITLE
TT-12347 fix management of custom keys in mdcb installations

### PR DESCRIPTION
**Pull request by sredxny**
## Related Issue

https://tyktech.atlassian.net/browse/TT-12347

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

___

### **PR Type**
Enhancement

___

### **Description**
- Changed the order of key search logic in the `SessionDetail` method of `DefaultSessionManager`.
- Moved `keyName` to be appended after generating token with `orgID`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>auth_manager.go</strong><dd><code>Modify key search order in `SessionDetail` method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gateway/auth_manager.go
<li>Changed the order of key search logic in <code>SessionDetail</code> method.<br> <li> Moved <code>keyName</code> to be appended after generating token with <code>orgID</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk/pull/6353/files#diff-311e18b071d244ed1615c0019215b633278679df373288b3451bcc5cf7c52c4e">+3/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: gateway/rpc_storage_handler.go [hunk 1]
```diff
@@ -1144,6 +1149,17 @@ func (r *RPCStorageHandler) ProcessKeySpaceChanges(keys []string, orgId string)
...
+func (r *RPCStorageHandler) deleteUsingTokenID(key, orgId string, resetQuota bool, status int) (int, error) {
+	if status == http.StatusNotFound {
+		id, err := storage.TokenID(key)
+		if err == nil {
+			_, status = r.Gw.handleDeleteKey(id, orgId, "-1", resetQuota)
+		}
+	}
+	return status, nil
+}
+
```
Comment (sredxny) [inline gateway/rpc_storage_handler.go:1162]: with this we will handle situations where a key is requested to be deleted using the base 64 representation, but it's stored using the custom id, Eg: `apikey-customID`

**Review by buger — APPROVED** [review]

**Pull request merged**